table.cpp

```diff
@@ -235,6 +260,30 @@ void table_t::stop()
             fprintf(stderr, "Creating osm_id index on %s\n", name.c_str());
             pgsql_exec_simple(sql_conn, PGRES_COMMAND_OK, (fmt("CREATE INDEX %1%_pkey ON %1% USING BTREE (osm_id) %2%") % name %
                 (table_space_index ? "TABLESPACE " + table_space_index.get() : "")).str());
+            if (srid != "4326") {
+                pgsql_exec_simple(
+                    sql_conn, PGRES_COMMAND_OK,
+                    (fmt("CREATE OR REPLACE FUNCTION %1%_valid()\n"
+                         "RETURNS TRIGGER AS $$\n"
+                         "BEGIN\n"
+                         "  IF ST_IsValid(NEW.way) THEN \n"
+                         "    RETURN NEW;\n"
+                         "  END IF;\n"
+                         "  RETURN NULL;\n"
+                         "END;"
+                         "$$ LANGUAGE plpgsql;") %
+                     name)
+                        .str());
+
+                pgsql_exec_simple(
+                    sql_conn, PGRES_COMMAND_OK,
+                    (fmt("CREATE TRIGGER %1%_valid BEFORE INSERT OR UPDATE\n"
+                         "  ON %1%\n"
+                         "    FOR EACH ROW EXECUTE PROCEDURE %1%_valid();") %
+                     name)
+                        .str());
```

Comment:
What about removing the trigger again?

Reply:
The trigger never needs to be explicitly removed. If reimporting, the entire table is dropped and the trigger goes with it. If importing more with --append, the trigger needs to be there.